metadata-ingestion/src/datahub/ingestion/source/vertexai.py

```diff
@@ -0,0 +1,795 @@
+                entityUrn=dataset_urn, aspect=SubTypesClass(typeNames=["Dataset"])
+            )
+        )
+
+        # Create a container for Project as parent of the dataset
+        container_key = ProjectIdKey(
+            project_id=self.config.project_id, platform=self.platform
+        )
+
+        mcps.append(
+            MetadataChangeProposalWrapper(
+                entityUrn=dataset_urn,
+                aspect=ContainerClass(
+                    container=container_key.as_urn(),
+                ),
+            )
+        )
+        yield from auto_workunit(mcps)
+
+    def _get_job_input_workunit(
+        self, job: VertexAiResourceNoun
+    ) -> Iterable[MetadataWorkUnit]:
+        """
+        Generate work units for the input data of a training job.
+        This method checks if the training job is an AutoML job and if it has an input dataset
+        configuration. If so, it creates a work unit for the input dataset.
+        """
+
+        if self._is_automl_job(job):
+            job_conf = job.to_dict()
+            if (
+                "inputDataConfig" in job_conf
+                and "datasetId" in job_conf["inputDataConfig"]
+            ):
+                # Create URN of Input Dataset for Training Job
+                dataset_id = job_conf["inputDataConfig"]["datasetId"]
+                logger.info(
+                    f" found a training job: {job.display_name} used input dataset id: {dataset_id}"
+                )
+
+                if dataset_id:
+                    yield from self._get_data_process_input_workunit(job, dataset_id)
+
+    def _get_data_process_input_workunit(
+        self, job: VertexAiResourceNoun, dataset_id: str
+    ) -> Iterable[MetadataWorkUnit]:
+        """
+        This method creates a work unit for the input dataset of a training job. It constructs the URN
+        for the input dataset and the training job, and then creates a DataProcessInstanceInputClass aspect
+        to link the input dataset to the training job.
+        """
+
+        # Create URN of Input Dataset for Training Job
+        dataset_name = self._make_vertexai_dataset_name(entity_id=dataset_id)
+        dataset_urn = builder.make_dataset_urn(
+            platform=self.platform,
+            name=dataset_name,
+            env=self.config.env,
+        )
+
+        dataset = self._search_dataset(dataset_id) if dataset_id else None
+        if dataset:
+            yield from self._get_dataset_workunit(dataset_urn=dataset_urn, ds=dataset)
+            # Create URN of Training Job
+            job_id = self._make_vertexai_job_name(entity_id=job.name)
+            mcp = MetadataChangeProposalWrapper(
+                entityUrn=builder.make_data_process_instance_urn(job_id),
+                aspect=DataProcessInstanceInputClass(inputs=[dataset_urn]),
+            )
+            logger.info(
+                f" found training job :{job.display_name} used input dataset : {dataset_name}"
```

Comment:
why are there a bunch of random spaces in this log message? also, the message can be way more clear

```suggestion
                f"Found input dataset {dataset_name} for training job {job.display_name}"
```

Reply:
changed.